File: poc/bank-simulator/src/main/java/eu/accesa/blinkpay/bank/dto/OfflineSyncResponse.java
package eu.accesa.blinkpay.bank.dto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

/**
 * Response from offline NFC transaction sync.
 * Returns the updated wallet balance and which transactions were accepted.
 * Duplicate transaction IDs are acknowledged but not re-applied.
 */
public record OfflineSyncResponse(
        UUID walletId,
        BigDecimal walletBalance,
        List<String> acceptedTransactionIds,
        List<String> duplicateTransactionIds
) {}

File: poc/mobile-client/app/src/main/java/eu/accesa/blinkpay/data/api/dto/OfflineSyncResponse.kt
package eu.accesa.blinkpay.data.api.dto

import java.math.BigDecimal

data class OfflineSyncResponse(
    val walletId: String,
    val walletBalance: BigDecimal,
    val acceptedTransactionIds: List<String>,
    val duplicateTransactionIds: List<String>,
)
